python/Ganga/Core/MonitoringComponent/Local_GangaMC_Service.py

```diff
@@ -81,16 +86,16 @@ def _execUpdateAction(self):
             if self.should_stop():
                 break
 
-            tpFreeThreads -= 1
             #setattr(threading.currentThread(), 'action', action)
             log.debug("Qin's size is currently: %d" % Qin.qsize())
             log.debug("%s running..." % threading.currentThread())
-
+            self._alive = True
```

Comment:
tbh, a global counter was even more confusing, especially when it wasn't ever reset to zero on ganga exit/cleanup. I can change the name but as you're talking about different classes and have an extra '_' I was thinking this was enough.